src/main/java/com/mongodb/kafka/connect/util/ConfigHelper.java

```diff
@@ -191,18 +192,17 @@ public static Optional<ConfigValue> getConfigByNameWithoutErrors(
     return Optional.empty();
   }
 
-  public static Config evaluateConfigValues(
-      final Config rawConfig, final AbstractConfig resolvedConfig) {
-    final Map<String, ?> evalValues = resolvedConfig.values();
-    rawConfig
-        .configValues()
-        .forEach(
-            configValue -> {
-              Object ev = evalValues.get(configValue.name());
-              if (ev != null) {
-                configValue.value(ev);
-              }
-            });
-    return rawConfig;
+  public static Map<String, String> evaluateConfigValues(
```

Comment:
This doesnt look right.  Why do we want to get from originals?  I thought values like in the above code is resolved configs which we want to use.

Reply:
Hi @jagadishmdb you are right. In the above code, **configValue.value(ev)** replaces the unresolved value with the resolved value. But the problem is configValue has another List property named "errorMessages". And that gets populated earlier. Please Check: MongoSinkConnector -> validate -> super.validate(connectorConfigs) call. errorCheckingPasswordValueValidator method adds that when super.validate is called with raw unresolved configs. 

So even though configValue.value is replaced with the resolved value, configValue.errorMessages is non empty at this point. Which eventually causes to throw the error down the line.  

Now, let's talk about the below code segment. Even though I am reading from originals, originals property actually has all the resolved values (sounds confusing!). The answer is in the constructor of the AbstractConfig class

`this.originals = resolveConfigVariables(configProviderProps, (Map<String, Object>) originals);` 

So the originals property has all the resolved values. And in my PR I am calling "super.validate" after raw config has been resolved, so the errorCheckingPasswordValueValidator does not get the opportunity to add that error message to the errorMessages property.